web/src/components/text-input/generic.tsx

```diff
@@ -8,20 +9,37 @@ import useStyles from './styles'
 import { useFocusStyle } from '../../styles/global'
 import { Flex } from '../layout/flex'
 import { Icon } from '../icon'
+import { onOverflowChanged } from '../scrollbar'
+import { genUid } from '../../utils'
 
-export interface TextInputProps extends AriaTextFieldOptions {
+import { TextAreaWithLabelProps, TextAreaWithoutLabelProps } from './text-area'
+import {
+  TextFieldWithLabelProps,
+  TextFieldWithoutLabelProps,
+} from './text-field'
+import {
+  SearchFieldWithLabelProps,
+  SearchFieldWithoutLabelProps,
+} from './search-field'
+
+export interface InputComponentProps extends AriaTextFieldOptions {
   id?: string
   isInvalid?: boolean
+  isDisabled?: boolean
+  isReadOnly?: boolean
+  isRequired?: boolean
   placeholder?: string
   description?: string
   validationHelp?: string
-  isRequired?: boolean
-  min?: number
-  max?: number
+  height?: number | string
+  width?: number | string
+  maxLength?: number
+  minLength?: number
   customLabels?: {
     required?: string
     optional?: string
   }
+  isNecessityLabel?: boolean
```

Comment:
add explanation what this does. e.g. "Whether to show a label a required/optional label, depending on isRequired." lit wasn't completely obvious to me.